src/common/messageBundles.js

```diff
@@ -0,0 +1,240 @@
+ * This approach makes use of the modelListeners in order for re-rendering to happen
+ * when the locale changes.
+ */
+fluid.defaults("gpii.app.messageBundles", {
+    gradeNames: ["fluid.modelComponent", "{that}.options.messageDistributorGrade"],
+
+    model: {
+        locale: null, // the defaultLocale will be used initially
+        messages: {}
+    },
+
+    defaultLocale: "en",
+
+    messageBundlesPath: "build/gpii-app-messageBundles.json",
+
+    messageBundles: "@expand:gpii.app.messageBundles.loadMessageBundles({that}.options.messageBundlesPath)",
+
+    modelListeners: {
+        locale: {
+            func: "{that}.updateMessages"
+        }
+    },
+
+    messageDistributorGrade: {
+        expander: {
+            funcName: "gpii.app.messageBundles.getMessageDistributorGrade",
+            args: [
+                "{that}.options.messageBundles",
+                "{that}.options.defaultLocale"
+            ]
+        }
+    },
+
+    invokers: {
+        updateMessages: {
+            funcName: "gpii.app.messageBundles.updateMessages",
+            args: [
+                "{that}",
+                "{that}.options.messageBundles",
+                "{that}.model.locale",
+                "{that}.options.defaultLocale"
+            ]
+        }
+    }
+});
+
+/**
+ * This function creates a `messageDistributor` grade which has a dynamically generated
+ * `distributeOptions` block. The name of the grade is returned as a result of this
+ * function and is applied to the `messageBundles` component. By using this raw dynamic grade
+ * mechanism this component is constructed before the other components. This is important
+ * because distributeOptions will be considered only at creation time. This means that the
+ * distributor should be created before any other components to which it distributes
+ * messages.
+ * @param messageBundles {Object} An object containing all i18n messages for all supported
+ * locales
+ * @param defaultLocale {String} A string representing the default locale.
+ * @return {String} The name of the distributor grade.
+ */
+gpii.app.messageBundles.getMessageDistributorGrade = function (messageBundles, defaultLocale) {
+    var defaultMessages = messageBundles[defaultLocale],
+        distributions = gpii.app.messageBundles.getMessageDistributions(defaultMessages);
+
+    fluid.defaults("gpii.app.messageDistributor", {
+        gradeNames: "fluid.component",
+        distributeOptions: distributions
+    });
+
+    return "gpii.app.messageDistributor";
+};
+
+/**
+ * Loads synchronously and parses the messageBundles file.
+ * @param messageBundlesPath {String} The path to the messageBundles file relative
+ * to the project directory.
+ * @return {Object} The parsed message bundles for the different locales.
+ */
+gpii.app.messageBundles.loadMessageBundles = function (messageBundlesPath) {
+    var resolvedPath = require("path").resolve(messageBundlesPath);
```

Comment:
This should allow module-relative references of the form %module/path and use https://docs.fluidproject.org/infusion/development/NodeAPI.html#fluidmoduleresolvepathpath

Reply:
OK, you have explained that because this executes in the renderer process, and also since this only references "built bundles" that the use of fluid.module.resolvePath is impossible/unnecessary